chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -129,8 +124,19 @@ def lock_pool(self):
         while not self.lock.i_am_locking():
             try:
                 self.lock.acquire(self.LOCK_ACQUIRE_TIMEOUT)
+                with open(self.lock.lock_file, 'w') as f:
+                    f.writelines(str(self.lock.pid))
             except LockTimeout:
-                pass
+                # prune locks owned by non-existent processes
+                with open(self.lock.lock_file, 'r') as f:
+                    contents = f.readlines()
+
+                assert len(contents) == 1 and contents[0].isdigit(), \
+                    'unexpected contents of lockfile %s: %s' % (self.lock.lock_file, contents)
+
+                # validate pid holding lock exists
+                if not pid_exists(int(contents[0])):
```

Comment:
in these situations it is possible that the unlock routine has not been completed and the lock stays held. then the subsequent Lock caller can check if the owner process exists and if not it is safe to break then reacquire the lock